### PR TITLE
Use origin of final HTTP response as document origin

### PR DESCRIPTION
**Pull request by jdm**
I discovered surprising problems when loading pages from http://joshmatthews.net that load same-origin images and try to use them as webgl textures. Since there's a redirect from joshmatthews.net to www.joshmatthews.net, the images are not considered same-origin in Servo because the document stores the origin for the original request's URL, rather than the origin of the response that is received after processing any redirections.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23887)

### CODE REVIEW

**Comment by asajeffrey** [comment]
This doesn't affect any test results? That's surprising! You can r=me

**Comment by jdm** [comment]
I made the PR so I could check the test results :)
@bors-servo try=wpt

**Comment by jdm** [comment]
That's kind of unnerving.

**Comment by asajeffrey** [comment]
Er yes, there are tests for similar-but-not-same-origin forwarding?

**Comment by jdm** [comment]
I added a test that failed before this change and now passes.
@bors-servo r=asajeffrey

File: tests/wpt/mozilla/tests/mozilla/document_origin_redirect.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/common/get-host-info.sub.js"></script>
+<body>
+<script>
+  async_test(function(t) {
+    var path = new URL("document_origin_redirect_subpage.html", document.location).pathname;
+    var CROSS_ORIGIN_URL = new URL(path, get_host_info().HTTP_REMOTE_ORIGIN);
+    var SAME_ORIGIN_URL = "/common/redirect.py?location=" + CROSS_ORIGIN_URL;
+      console.log(SAME_ORIGIN_URL);
```
Comment (gterzian) [inline tests/wpt/mozilla/tests/mozilla/document_origin_redirect.html]: Looks like the wpt linter only lints tests in the wpt folder...

**Pull request merged**